### ISSUE TITLE
Null continuation doesn't escape from calling context.

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap--Build-Your-Own-Blocks#1212]
From Jonathan50 via forum:

Continuations (and so catch/throw, and so the multibranched conditionals library) don't “escape” when there is nothing to do (the continuation is empty).
So if I have a cases block with nothing after it then all the cases are tested…

Confirming:  This script
![cases-test](https://cloud.githubusercontent.com/assets/3885074/15090655/98aff916-13e3-11e6-8617-5cd9f87878da.png)
should just say "b" but actually says b-c-d unless another command follows the CASES block.

**Comment by brianharvey** [comment]
Note:  I have temporarily kludged around the problem in the cases library by adding a no-op command after the catch.

**Comment by jmoenig** [comment]
Good catch, will fix. Thanks!

**Comment by jmoenig** [comment]
I'm confused. I know that there is an issue if you try to catch the very end of a script unless you specifically tag it. But I can't reproduce the behavior you describe. Here's your test case:

http://snap.berkeley.edu/snapsource/snap.html#present:Username=jens&ProjectName=ctest

it says 'b' and then stops. Doesn't it for you?

**Comment by brianharvey** [comment]
That's because you missed my comment above saying I edited the CASES library to add a no-op (the HIDE VARIABLE block) at the end.  Edit the CASES block, remove that no-op, and then try.

**Comment by jmoenig** [comment]
Du-oh. Thanks!

**Comment by jmoenig** [comment]
Okay, can you please test my fix for null continuations in the dev version:
    http://snap.berkeley.edu/snapsource/dev/snap.html
Thanks!

**Comment by brianharvey** [comment]
Yay!  It works.  It didn't break the thread demo either.  :-)

**Comment by jmoenig** [comment]
Thanks for the quick test. The fix is really simple: A null continuation terminates the thread, same as ```stop this script```.

**Comment by brianharvey** [comment]
Seems sad that you had to special case it.  An empty stack should really be just like any stack.  (And while you're at it, I want world peace.)

**Comment by jmoenig** [comment]
I already did special case null continuations, but only as noops, which was clearly wrong, all I did now was to change it from "doYield" to "doStop". In a perfect Scheme world, of course, scripts would be represented as lists, so I could just "set" it to the end of that list. In a perfect blocks language the blocks themselves are the ... actually I don't know anything about perfect languages. But Yay for world peace!
